src/translate/java/org/javarosa/xform/schema/FormInstanceLoader.java

```diff
@@ -45,29 +45,12 @@ public static FormDef loadInstance(InputStream formInput,
 
     public static FormDef loadInstance(FormDef formDef,
                                        InputStream instanceInput) throws IOException {
-        FormInstance savedModel;
-        savedModel = XFormParser.restoreDataModel(instanceInput, null);
-
-        // get the root of the saved and template instances
-        TreeElement savedRoot = savedModel.getRoot();
-        TreeElement templateRoot =
-                formDef.getInstance().getRoot().deepCopy(true);
-
-        // weak check for matching forms
-        if (!savedRoot.getName().equals(templateRoot.getName()) ||
-                savedRoot.getMult() != 0) {
-            System.out.println("Instance and template form definition don't match");
+        FormInstance formInstance = XFormParser.restoreDataModel(instanceInput, null);
+        try {
+            return XFormParser.loadXmlInstance(formDef, formInstance);
+        } catch (RuntimeException e) {
```

Comment:
I think it would be ok to change behaviour here and not catch this exception here but let this bubble up to top level Formplayer classes just like the `XFormParser.loadXmlInstance`

Reply:
Won't that make the cli crash?

Reply:
yeah though I think it's better to fail hard and crash rather than causing unseen issues. Like currently if there is an error in this part of the code, we will hardly get to know about them unless we are proactively looking out for the system log.

Reply:
22fc0f38b